packages/apps/desktop-viewer-test/src/frontend/components/App.tsx

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useDesktopViewerInitializer } from "@itwin/desktop-viewer-react";
+import { Router } from "@reach/router";
+import React, { useEffect, useState } from "react";
+
+import { ViewerSettings } from "../../common/ViewerConfig";
+import {
+  addRecentOnline as addRecentOnlineClient,
+  addRecentSnapshot as addRecentSnapshotClient,
+  getUserSettings,
+  SettingsContext,
+} from "../services/SettingsClient";
+import {
+  DownloadRoute,
+  HomeRoute,
+  IModelsRoute,
+  ITwinsRoute,
+  SnapshotRoute,
+} from "./routes";
+
+const App = () => {
+  const initialized = useDesktopViewerInitializer();
+  const [settings, setSettings] = useState<ViewerSettings>();
+
+  useEffect(() => {
+    if (initialized) {
+      void getUserSettings().then((userSettings) => {
+        setSettings(userSettings);
+      });
+    }
+  }, [initialized]);
+
+  const addRecentSnapshot = async (path: string) => {
```

Comment:
should this and addRecentOnline be memoized with useCallback?

Reply:
updated